Add DSM 7 support to synology.md (#1673)
openhab/openhab-docs: installation/synology.md

title: Synology DiskStation
---

# Synology DiskStation DSM 7.0 and higher

> The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages. For versions of DSM before 7.0 please refer to the second half of this page.

Reviewer: Each sentence in Markdown should be placed on a new line so it easier to review and compare changes.

```suggestion
The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages.
For versions of DSM before 7.0 please refer to the second half of this page.
```

Author: Tnx for the review, very useful feedback.

## Prerequisite - prepare the Synology OS for openhab

> Synology NAS is based on unix but it has limitations since it is not a full server. There are some shortcuts you have to take to make it work. For the use of openHAB you need some preparation steps. For this create a ssh session into your NAS. For reference how to do that: I found this [youtube tutorial](https://www.youtube.com/watch?v=BCCIMRbAUp8).

Reviewer: My Synlogy NAS has a Linux kernel.

```suggestion
Synology NAS is based on Linux but it has limitations since it is not a full server.
There are some shortcuts you have to take to make it work.
For the use of openHAB you need some preparation steps.
For this create a ssh session into your NAS.
For reference how to do that: I found this [youtube tutorial](https://www.youtube.com/watch?v=BCCIMRbAUp8).
```

When you are in the secured shell first thing to do is to create the user openhab.
```bash
sudo synouser --add openhab habopen
sudo synogroup --add openhab openhab
```

> If you are going to use an usb like a z-wave dongle add another group for openhab and change the port access for ttyACM0 if that is the port for your usb device

Reviewer: ```suggestion
If you are going to use an usb like a Z-Wave dongle add another group for openHAB and change the port access for ttyACM0 if that is the port for your USB device
```

Please also use the proper capitalization for brands/abbreviations/products in the other parts:

* Docker
* openHAB
* Windows

I created #1671 to correct similar issues changes recently. :wink:

Author: Ok. clear

title: Synology DiskStation
---

# Synology DiskStation DSM 7.0 and higher

The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages. 
For versions of DSM before 7.0 please refer to the second half of this page.

> ## Prerequisite - prepare the Synology OS for OpenHAB

Reviewer: ```suggestion
## Prerequisite - prepare the Synology OS for openHAB
```

Synology NAS is based on Linux but it has limitations since it is not a full server.
There are some shortcuts you have to take to make it work.  

> For the use of OpenHAB you need some preparation steps.

Reviewer: ```suggestion
For the use of openHAB you need some preparation steps.
```

For this create a ssh session into your NAS.
For reference how to do that: I found this [youtube tutorial](https://www.youtube.com/watch?v=BCCIMRbAUp8).
When you are in the secured shell first thing to do is to create the user openhab.
```bash
sudo synouser --add openhab habopen
sudo synogroup --add openhab openhab
```
If you are going to use an USB like a Z-Wave dongle add another group for openhab and change the port access for ttyACM0 if that is the port for your USB device
```bash
sudo synogroup --add dialout openhab
sudo chown root:dialout /dev/ttyACM0
sudo chmod g+rw /dev/ttyACM0
```
Then for installation purposes run the following command:
```bash
sudo id openhab
```
The result will something like
```bash
uid=1032(openhab) gid=100(users) groups=100(users),65537(dialout),65539(openhab)
```
In this case we need the 1032 as the user and the 65539 as the group. Write them down.

## Docker

> DSM 7 does not longer support java8 directly and therefor OpenHAB has to run in a Docker container. 

Reviewer: ```suggestion
DSM 7 does no longer support Java 11 directly and therefor openHAB has to run in a Docker container. 
```

And that also makes the installation easier to maintain. 

> It works out of the box a bit different then the normal Docker installation as described in de OpenHAB documententation.

Reviewer: ```suggestion
It works out of the box a bit different then the normal Docker installation as described in de openHAB documententation.
```

Docker is a containerization platform and is used to run lightweight containers. 
These containers require a very little amount of memory and system resources to run. 
Synology NAS has official support for Docker.
To use Docker, you need to install the Docker app from the Synology Web GUI.

- Login and open the DiskStation Manager.
- Go to Main Menu → Package Center.
- Search for "Docker" and install.
- Click "Open".
- Go to "Registry", here are all the available Docker images and search for openhab.

> - Choose the "openhab/openhab" image and click "Download". Select "latest" since that is the latest stable version of OpenHAB. The Docker image is added to "Image"

Reviewer: ```suggestion
- Choose the "openhab/openhab" image and click "Download". Select "latest" since that is the latest stable version of openHAB. The Docker image is added to "Image"
```

### Advanced Settings

- Enable auto-restart

### Volume

- Click on "Add Folder" and "Create Folder" under the folder "docker" with the name "openHAB".
- Create the folder "userdata" in the folder "openHAB" 
- Create the folder "conf" in the folder "openHAB" 
- Create the folder "addons" in the folder "openHAB" 
- Now select "userdata" and click on "Select"
- Change the mount path to "/openhab/userdata" this will connect the Docker containers path "/openhab/userdata" to your Synology's local path "docker/openHAB/userdata" where you can reach it from outside the Docker.
- Click on "Add Folder", select "docker/openHAB/conf", click on select and change the mount path to "/openhab/conf".
- Repeat this for addons.

### Network

- Select 'host' if you are planning to use the IP stack in bindings or if you are not sure what this does.

### Port Settings

- No change

### Links

- Nothing to do here

### Environment

- In the list find the variable "GROUP_ID" and put your previous written down id of the group openhab here, in this tutorial it is 65539.
- In the same list locate "USER_ID" and fill the id of the user openhab, in this tutorial 1032.
- Click on "Apply".

### Summary 

> Here you can check if all changes where set correctly for your OpenHAB container, enable "Run this container after the wizard is finished" and press "Done".

Reviewer: ```suggestion
Here you can check if all changes where set correctly for your openHAB container, enable "Run this container after the wizard is finished" and press "Done".
```

> OpenHAB is now running on your NAS. 

Reviewer: ```suggestion
openHAB is now running on your NAS. 
```

Give it a few minutes to start up.

## Shells

If you select the container running and press "Details" you will end up on  a windows where you get more information on this running container. 
One particular menu item you have to check is "Terminal". 

> In here you see the name of the container left from the black field and if you press that you'll get the Karaf terminal for OpenHAB.

Reviewer: ```suggestion
In here you see the name of the container left from the black field and if you press that you'll get the Karaf terminal for openHAB.
```

If you click on "Create' it creates a bash terminal, click on bash and you'll have access to a shell in the container.

> By now the container is up and running, goto "http://[your-NAS-ip-or-host-name]:8080" and the welcome to OpenHAB.

Reviewer: ```suggestion
By now the container is up and running, goto "http://[your-NAS-ip-or-host-name]:8080" and the welcome to openHAB.
```

> If you use p.e. a Z-Wave dongle on ttyACM0 you can now add the Z-Wave binding. Once OpenHAB has been configured under the binding Z-Wave add a controller manually and change the port to /dev/ttyACM0. The Z-Wave controller gets online and you will discover the Z-Wave devices. If this is not working, please verify you enabled high privilege.

Reviewer: ```suggestion
If you use p.e. a Z-Wave dongle on ttyACM0 you can now add the Z-Wave binding. Once openHAB has been configured under the binding Z-Wave add a controller manually and change the port to /dev/ttyACM0. The Z-Wave controller gets online and you will discover the Z-Wave devices. If this is not working, please verify you enabled high privilege.
```

::: tip Note
The device path (`/dev/ttyACM0`) or container name (`openhab`) could be different in your system, please modify the commands accordingly.
:::

---

# Older Synology DiskStations till DSM 6

> The [DiskStation by Synology](https://www.Synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages. 

Reviewer: ```suggestion
The [DiskStation by Synology](https://www.synology.com/en-us/dsm) is a famous NAS server solution for your home, allowing the installation of additional packages.
```

> We are proud to be able to provide an [OpenHAB Synology package](https://github.com/openhab/openhab-syno-spk).

Reviewer: ```suggestion
We are proud to be able to provide an [openHAB Synology package](https://github.com/openhab/openhab-syno-spk).
```

## Prerequisite - Install Java

> Current releases of OpenHAB require Oracle Java due to the generation of certificates based on [EC](https://en.wikipedia.org/wiki/Elliptic_curve_cryptography) which is not available in free Java implementations.

Reviewer: ```suggestion
Current releases of openHAB require Oracle Java due to the generation of certificates based on [EC](https://en.wikipedia.org/wiki/Elliptic_curve_cryptography) which is not available in free Java implementations.
```

The filename and format of the uploaded file will be checked before installation.

**Armada 370/XP** systems running DSM 6 require hard float ABI rather than the soft float used on DSM 5.

> If there is not a Java8 package in the Package Center, your machine may not be supported by the [Synology package](https://www.Synology.com/en-us/dsm/packages/Java8).

Reviewer: ```suggestion
If there is not a Java8 package in the Package Center, your machine may not be supported by the [Synology package](https://www.synology.com/en-us/dsm/packages/Java8).
```

> The best option for older unsupported Synology models such as **PowerPC** and **ARMv5** is to include the [PC-Loadletter](https://pcloadletter.co.uk/2011/08/23/java-package-for-Synology/) Repo.

Reviewer: ```suggestion
The best option for older unsupported Synology models such as **PowerPC** and **ARMv5** is to include the [PC-Loadletter](https://pcloadletter.co.uk/2011/08/23/java-package-for-synology/) Repo.
```

## Installation

Download the latest SPK package: [Releases](https://github.com/openhab/openhab-syno-spk/releases)

> The SPK is a wrapper to download the latest OpenHAB release and does not contain OpenHAB itself.

Reviewer: ```suggestion
The SPK is a wrapper to download the latest openHAB release and does not contain openHAB itself.
```

## Configuration

> The location of configuration files by OpenHAB differs depending on wether or not a shared folder named `public` exists:

Reviewer: ```suggestion
The location of configuration files by openHAB differs depending on wether or not a shared folder named `public` exists:
```

The administrator can also create a public shared folder if desired.

Synology does not document which models will automatically create a public folder.

> Refer to the Synology knowlegde base article on [Shared Folder](https://www.Synology.com/en-us/knowledgebase/DSM/help/DSM/AdminCenter/file_share_desc) to learn more.

Reviewer: ```suggestion
Refer to the Synology knowlegde base article on [Shared Folder](https://www.synology.com/en-us/knowledgebase/DSM/help/DSM/AdminCenter/file_share_desc) to learn more.
```

## Logging

> The OpenHAB log files can be found here:

Reviewer: ```suggestion
The openHAB log files can be found here:
```
